File: 2022/05/17/powerOfFour.py
"""
https://leetcode.com/problems/power-of-four/
Author: Andrew Jarombek
Date: 5/17/2022
"""


def is_power_of_four(n: int) -> bool:
    if n <= 0:
        return False

    while n > 1:
        print(n)
        if n % 4 != 0:
            return False
        n //= 4

    return True

File: 2022/07/15/java/PowerOfFour.java
/**
 * https://leetcode.com/problems/power-of-four/
 * @author Andrew Jarombek
 * @since 7/15/2022
 */

public class PowerOfFour {
    public static boolean isPowerOfFour(int n) {
        if (n <= 0)
            return false;

        while (n > 1) {
            if (n % 4 == 0) {
                n /= 4;
            } else {
                return false;
            }
        }

        return true;
    }

    public static void main(String[] args) {
        assert !isPowerOfFour(0);
        assert isPowerOfFour(16);
    }
}
